maintaining-and-tuning/monitoring.md

```diff
@@ -0,0 +1,152 @@
+# Monitoring
+This page outlines how to set up monitoring for the GRR components. GRR keep tracks of many metrics
+which can be used to create charts about GRR's performance, health, and workload.
+
+When `Monitoring.http_port` is configured, each GRR component exports metrics at 
+`http://<host>:<port>/metrics`. The export is human-readable plain-text and follows the 
+[Open Metrics standard](https://openmetrics.io/). More importantly however,
+[Prometheus](https://prometheus.io) can parse the metrics.
+
+
+## Example Setup
+This example will walk you through a basic Prometheus setup. For this example, the GRR Frontend,
+Worker, and Admin UI will be launched on your local machine. Please refer to
+[Installing GRR Server](../installing-grr-server) for a real-world setup.
+
+1. Install GRR, for example from [pip](../installing-grr-server/from-released-pip.html).
+
+1. Run the GRR components locally. Because the monitoring ports would clash, assign a custom port
+to each service. In a regular setup, where each component runs on a dedicated host, the custom ports
+are not needed. Execute each of the three commands in a separate terminal:
```

Comment:
There is also the config option http_port_max to define a port range used for monitoring, specifically to avoid those issues. Would be worth mentioning here?

Reply:
Usage of a port range is not viable when using Prometheus. With a port range, the exact port that a services listens on is depending on the startup order. Prometheus requires a fixed association.

Reply:
Added clarification.